src/ProjectTemplates/Web.ProjectTemplates/content/Api-CSharp/Program.cs

```diff
@@ -0,0 +1,34 @@
+#if (NativeAot)
+using System.Text.Json.Serialization;
+#endif
+using Company.ApiApplication1;
+
+var builder = WebApplication.CreateBuilder(args);
+builder.Logging.AddConsole();
+
+#if (NativeAot)
+builder.Services.ConfigureHttpJsonOptions(options =>
+{
+    options.SerializerOptions.AddContext<AppJsonSerializerContext>();
+});
+
+#endif
+var app = builder.Build();
+
+var sampleTodos = TodoGenerator.GenerateTodos().ToArray();
+
+var todosApi = app.MapGroup("/todos");
+todosApi.MapGet("/", () => sampleTodos);
+todosApi.MapGet("/{id}", (int id) =>
+    sampleTodos.FirstOrDefault(a => a.Id == id) is { } todo
+        ? Results.Ok(todo)
+        : Results.NotFound());
```

Comment:
@captainsafia @davidfowl -  How is code like this going to allow our minimal API source generator to infer that we need to add `Todo` to the JSON Source Generation? Will we look through the delegate and find all places that call `Results.X`? We won't be able to tell by the Delegate's return type, because here that will just be `IResult`, and not `IResult<Todo>`

Reply:
I spoke to @davidfowl about this recently and IIRC there isn't a firm plan for the SG to do this as yet. For now, the app would be responsible for ensuring returned types are declared on a JSON serializer context that is configured in DI. We talked about analyzers and code-fixers that could make the process of keeping these up to date easier, and of course ideally the minimal API SG would do this automatically as you suggest, but as I understand it that capability will not be there initially.

Reply:
That's right. There's currently no plan to call the JSON source generator from the minimal API one. All we have are vague ideas. I assume the JSON source generator would need to be refactored to do this, but it would be good to attempt this once we have the baseline functionality checked in.

Maybe @brunolins16 can investigate once we're done with the core JSON work.

Reply:
If you're interested though, I wrote some ghetto logic for this here:

https://github.com/davidfowl/uController/blob/7c1e8f3db1c6a20738eff792386696ff89d4d7e7/src/uController.SourceGenerator/uControllerGenerator.cs#L366

It was for a different purpose though, identifying the metadata for open api.